Scripts/RemediationScripts/Remediate-EnableAADForKubernetesService.ps1

```diff
@@ -0,0 +1,523 @@
+            $kubernetesServicesWithoutAADEnabled += $kubernetesServiceResource | Select-Object @{N='ResourceID';E={$resourceId}},
+                                                                                   @{N='ResourceGroupName';E={$resourceGroupName}},
+                                                                                   @{N='ResourceName';E={$resourceName}},
+                                                                                   @{N='Location';E={$location}},
+                                                                                   @{N='IsRBACEnabled';E={$isRBACEnabled}},
+                                                                                   @{N='IsAADEnabled';E={$isAADEnabled}}
+        }
+        catch
+        {
+            $kubernetesServicesSkipped += $kubernetesServiceResource
+            Write-Host "Error fetching Kubernetes Services configuration: Resource ID - $($resourceId), Resource Group Name - $($resourceGroupName), Resource Name - $($resourceName). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+        }
+    }
+
+    $totalKubernetesServicesWithoutAADEnabled = ($kubernetesServicesWithoutAADEnabled | Measure-Object).Count
+
+    if ($totalKubernetesServicesWithoutAADEnabled -eq 0)
+    {
+        Write-Host "No Kubernetes Service found with AAD disabled. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+
+    Write-Host "`n $($totalKubernetesServicesWithoutAADEnabled) out of $($totalKubernetesServices) Kubernetes Service(s) with AAD disabled." -ForegroundColor $([Constants]::MessageType.Update)
+
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\EnableAADForKubernetesServices"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+
+    Write-Host $([Constants]::DoubleDashLine)
+
+    if (-not $DryRun)
+    {
+        $colsProperty = @{Expression={$_.ResourceId};Label="Resource ID";Width=40;Alignment="left"},
+                        @{Expression={$_.ResourceGroupName};Label="Resource Group Name";Width=20;Alignment="left"},
+                        @{Expression={$_.ResourceName};Label="Resource Name";Width=20;Alignment="left"},
+                        @{Expression={$_.Location};Label="Location";Width=20;Alignment="left"},
+                        @{Expression={$_.isRBACEnabled};Label="Is RBAC enabled on Kubernetes Cluster?";Width=20;Alignment="left"},
+                        @{Expression={$_.isAADEnabled};Label="Is AAD enabled on Kubernetes Cluster?";Width=20;Alignment="left"}
+
+        $kubernetesServicesWithoutAADEnabled | Format-Table -Property $colsProperty -Wrap
+
+        Write-Host $([Constants]::DoubleDashLine)
```

Comment:
This code is duplicate as we are taking backup in both case dry and wet run lets move this out.